modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/exportimport/data/handler/DEDataDefinitionFieldLinkStagedModelDataHandler.java

```diff
@@ -136,29 +150,49 @@ protected void doImportStagedModel(
 			portletDataContext.getImportDataStagedModelElement(
 				deDataDefinitionFieldLink);
 
+		String className = deDataDefinitionFieldLinkElement.attributeValue(
+			"link-class-name");
+
 		importedDEDataDefinitionFieldLink.setGroupId(
 			portletDataContext.getScopeGroupId());
 		importedDEDataDefinitionFieldLink.setCompanyId(
 			portletDataContext.getCompanyId());
 		importedDEDataDefinitionFieldLink.setClassNameId(
-			_portal.getClassNameId(
-				deDataDefinitionFieldLinkElement.attributeValue(
-					"link-class-name")));
+			_portal.getClassNameId(className));
 		importedDEDataDefinitionFieldLink.setDdmStructureId(ddmStructureId);
 
-		DDMStructure ddmStructure = _ddmStructureLocalService.getDDMStructure(
-			ddmStructureId);
+		if (className.equals(DDMStructureLayout.class.getName())) {
+			long linkedStructureId = GetterUtil.getLong(
+				deDataDefinitionFieldLinkElement.attributeValue(
+					"linked-structure-id"));
+
+			linkedStructureId = MapUtil.getLong(
+				ddmStructureIds, linkedStructureId, linkedStructureId);
 
-		DDMStructureVersion structureVersion =
-			ddmStructure.getStructureVersion();
+			DDMStructure ddmStructure =
+				_ddmStructureLocalService.getDDMStructure(linkedStructureId);
 
-		DDMStructureLayout structureLayout =
-			_ddmStructureLayoutLocalService.
-				getStructureLayoutByStructureVersionId(
-					structureVersion.getStructureVersionId());
+			DDMStructureVersion structureVersion =
+				ddmStructure.getStructureVersion();
```

Comment:
@mbowerman could you rename from `structureVersion` to `ddmStructureVersion`? Thanks!

Reply:
@carolmariaabb Done